trading_calendars/exchange_calendar_xphs.py

```diff
@@ -0,0 +1,425 @@
+#
+# Copyright 2019 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time, timedelta
+from itertools import chain
+import pandas as pd
+from pandas.tseries.holiday import (
+    Holiday,
+    GoodFriday,
+)
+from pytz import timezone
+
+from .common_holidays import (
+    european_labour_day,
+    new_years_day,
+    maundy_thursday,
+    christmas,
+    christmas_eve,
+    new_years_eve,
+    all_saints_day,
+    chinese_lunar_new_year_dates,
+)
+from .trading_calendar import (
+    TradingCalendar,
+    HolidayCalendar,
+    MONDAY,
+    FRIDAY,
+)
+
+
+def only_friday(dt):
+    """
+    Only keeps the holidays that fall on a Friday.  Useful
+    for defining holidays that add a Friday to make a 4-day weekend
+    when falling on a Thursday.
+    """
+    return dt[dt.weekday == FRIDAY]
+
+
+def previous_monday(dt):
+    """
+    Returns previous monday on or before the given date.
+    """
+    while dt.weekday() != MONDAY:
+        dt -= timedelta(days=1)
+    return dt
+
+
+# All pre-2011 holidays are pre-computed, so we define Holidays starting
+# in 2011.
+NewYearsDay = new_years_day(start_date='2011')
+
+# Chinese New Year's only observed after 2011
+ChineseNewYear = chinese_lunar_new_year_dates
+
+ChineseNewYearAfter2011 = ChineseNewYear[ChineseNewYear.year > 2011]
+
+ArawNgKagitingan = Holiday(
+    'Araw Ng Kagitingan',
+    month=4,
+    day=9,
+    start_date='2011',
+)
+
+MaundyThursday = maundy_thursday(start_date='2011')
+
+LabourDay = european_labour_day(start_date='2011')
+
+IndependenceDay = Holiday(
+    'Independence Day',
+    month=6,
+    day=12,
+    start_date='2011',
+)
+
+NinoyAquinoDay = Holiday(
+    'Ninoy Aquino Day',
+    month=8,
+    day=21,
+    start_date='2011',
+)
+
+NationalHeroesDay = Holiday(
+    'National Heroes Day',
+    month=8,
+    day=31,
+    observance=previous_monday,
```

Comment:
I'm not positive if this works, but another way to do this might be to, instead of using `observance`, use `offset=DateOffset(weekday=MO(-1))`, where `MO` stands for Monday (`from dateutil.relativedelta import MO`)

Reply:
I forgot about DateOffset!  Made this change